dashboard/app/models/unit_group.rb

```diff
@@ -85,13 +85,30 @@ def self.file_path(name)
   def self.load_from_path(path)
     serialization = File.read(path)
     hash = JSON.parse(serialization)
+    UnitGroup.seed_from_hash(hash)
+  end
 
+  def self.seed_from_hash(hash)
     unit_group = UnitGroup.find_or_create_by!(name: hash['name'])
     unit_group.update_scripts(hash['script_names'], hash['alternate_scripts'])
     unit_group.properties = hash['properties']
-    unit_group.save!
 
+    # add_course_offering creates the course version
     CourseOffering.add_course_offering(unit_group)
+    course_version = unit_group.course_version
+
+    if course_version && hash['resources']
+      resources_imported = hash['resources'].map do |resource_data|
+        resource_attrs = resource_data.except('seeding_key')
+        resource_attrs['course_version_id'] = course_version.id
+        resource = Resource.find_or_create_by(key: resource_attrs['key'])
+        resource.update!(resource_attrs)
+        resource
```

Comment:
I would suggest this pattern to avoid writes when nothing has changed:
```suggestion
        resource = Resource.first_or_initialize_by(key: resource_attrs['key'])
        resource.assign_attributes(resource_attrs)
        resource.save! if resource.changed?
```

Reply:
Thanks!